libs/core/src/lib/form/form-control/form-control.directive.ts

```diff
@@ -30,51 +29,58 @@ export class FormControlDirective extends AbstractFdNgxClass {
     @Input()
     compact: boolean = false;
 
-
     @Input()
     type: string;
 
+    /** user's custom classes */
+    @Input()
+    class: string;
+
+    @applyCssClass
+    /** CssClassBuilder interface implementation
+     * function must return single string
+     * function is responsible for order which css classes are applied
+     */
+    buildComponentCssClass(): string {
+        return [
+            this.state ? 'is-' + this.state : '',
+            this.type === 'radio' ? 'fd-radio' : '',
+            this._getElementTag() === 'input' ? 'fd-input' : '',
+            this._getElementTag() === 'textarea' ? 'fd-textarea' : '',
+            this._getElementTag() === 'select' ? 'fd-form-select' : '',
+            this.compact && this._getElementTag() === 'input' ? 'fd-input--compact' : '',
+            this.compact && this._getElementTag() === 'textarea' ? 'fd-textarea--compact' : '',
+            this.compact && this._getElementTag() === 'select' ? 'fd-form-select--compact' : '',
+            this.class
+        ].filter(x => x !== '').join(' ');
+    }
```

Comment:
Can we have some method there?
```suggestion

    buildComponentCssClass(): string {
        return [
            this.state ? 'is-' + this.state : '',
            this._getFormClass(),
            this.compact ? (this._getFormClass() + '--compact') : '',
            this.class
        ].filter(x => x !== '').join(' ');
    }
        
    private _getFormClass(): string {
        let class: string = '';
        switch (this._getElementTag()) {
            case: 'input': 
                class = 'fd-input';
                break;
            case: 'select': 
            ...
        }
    }
```